### PR TITLE
zarr ckpt to torch_dist ckpt converter

### PR DESCRIPTION
**Pull request by dimapihtar**
# What does this PR do ?

Adds a script to convert zarr ckpt format to torch distributed ckpt format.

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

**Comment by dimapihtar** [comment]
jenkins

File: examples/nlp/language_modeling/megatron_zarr_ckpt_to_torch_dist.py
Comment (ericharper) [inline examples/nlp/language_modeling/megatron_zarr_ckpt_to_torch_dist.py:1]: Let's move this to scripts/checkpoint_converters

File: examples/nlp/language_modeling/megatron_zarr_ckpt_to_torch_dist.py [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        scaler = GradScaler(
+            init_scale=cfg.model.get('native_amp_init_scale', 2 ** 32),
+            growth_interval=cfg.model.get('native_amp_growth_interval', 1000),
+            hysteresis=cfg.model.get('hysteresis', 2),
+        )
+    plugins.append(PipelineMixedPrecisionPlugin(precision=cfg.trainer.precision, device='cuda', scaler=scaler))
+    # Set precision None after precision plugins are created as PTL >= 2.1 does not allow both
+    # precision plugins and precision to exist
+    cfg.trainer.precision = None
+    trainer = Trainer(plugins=plugins, strategy=strategy, **cfg.trainer)
```
Comment (yaoyu-33) [inline examples/nlp/language_modeling/megatron_zarr_ckpt_to_torch_dist.py:146]: can you use `MegatronTrainerBuilder` https://github.com/NVIDIA/NeMo/blob/main/scripts/checkpoint_converters/convert_gemma_hf_to_nemo.py#L215

File: examples/nlp/language_modeling/megatron_zarr_ckpt_to_torch_dist.py [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+    app_state.pipeline_model_parallel_rank = parallel_state.get_pipeline_model_parallel_rank()
+    app_state.tensor_model_parallel_rank = parallel_state.get_tensor_model_parallel_rank()
+
+    # check for distributed checkpoint
+    dist_ckpt_dir = os.path.join(args.checkpoint_folder, args.checkpoint_name)
+    if os.path.isdir(dist_ckpt_dir):
+        checkpoint_path = dist_ckpt_dir
+    else:
+        # legacy checkpoint needs model parallel injection
+        checkpoint_path = inject_model_parallel_rank(os.path.join(args.checkpoint_folder, args.checkpoint_name))
```
Comment (yaoyu-33) [inline examples/nlp/language_modeling/megatron_zarr_ckpt_to_torch_dist.py:169]: I don't think this script will support legacy checkpoints. The legacy checkpoints have different key names, so need to rename the keys. So I feel you no longer need checkpoint folder and checkpoint name, probably just a path to zarr checkpoint

**Review by yaoyu-33 — APPROVED** [review]

**Pull request merged**